File: schemas/social-media/convex/mentions.ts
// mentions: tracks user mentions within posts for notifications and lookups.
// See README.md for full design rationale.

import { defineTable } from "convex/server";
import { v } from "convex/values";

export const mentions = defineTable({
  postId: v.id("posts"),
  mentionedUserId: v.id("users"),
})
  .index("by_post_id_mentioned_user_id", ["postId", "mentionedUserId"])
  .index("by_mentioned_user_id_creation_time", ["mentionedUserId", "_creationTime"]);

File: schemas/social-media/spacetimedb/mentions.rs
// mentions: user mentions within posts.
// See README.md for full design rationale.

use spacetimedb::Timestamp;

#[spacetimedb::table(name = mentions, public)]
pub struct Mention {
    #[primary_key]
    pub id: String, // UUID
    // Composite unique: (post_id, mentioned_user_id) — enforce in reducer logic.
    #[index(btree)]
    pub post_id: String, // UUID — FK → posts.id (cascade delete)
    // Composite index: (mentioned_user_id, created_at) — not supported in SpacetimeDB; document only.
    #[index(btree)]
    pub mentioned_user_id: String, // UUID — FK → users.id (cascade delete)
    pub created_at: Timestamp,
}
